### ISSUE TITLE
Array, HashMap and List initialisation syntax

### ISSUE THREAD

**Issue by SamTebbs33** [ash-lang/ash#37]
# Arrays
Array initialisation syntax should be changed to

```
var array = [1, 2, 3, 4] // Static initialisation
var array2 = new [int, 10] // Initialised with a type an size
```
This would then free up the `{` and `}` characters for use with list initialisation

# Lists
```
var list = {1, 2, 3} // Static initialisation, Inferred as List<Integer>
var = new {Integer, 10} // A list initialised with a type and a size
```

The type of list that should be instantiated when using the list syntax is under consideration, proposals are as follows:
* When using static initialisation, the type of list should be inferred as `LinkedList` as this is the most commonly used list type.
* When using the second type of initialisation, if a size is included then an `ArrayList` should be instantiated, otherwise a `LinkedList` is used.

# HashMaps
HashMap initialisation could be achieved with braces but with colons separating keys and values.

```
var hashMap = {"key1" : "val1", "key2" : "val2"} // Inferred as HashMap<String, String>
var val1 = hashMap["key1"]
```

**Comment by SamTebbs33** [comment]
Updated array init syntax in 91f907c

**Comment by SamTebbs33** [comment]
Closed by 3667f2a

**Comment by SamTebbs33** [comment]
List size init generation was added by fdffc36
